modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala

```diff
@@ -1118,9 +1120,50 @@ abstract class BspTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
+  test("bloop projects are initialised properly even for faulty using directives") {
+    val invalidScalaVersionDirective   = "//> using scala 3.1.2"
+    val unfetchableDependencyDirective = "//> using lib \"no::lib:123\""
+    Seq(invalidScalaVersionDirective, unfetchableDependencyDirective)
+      .foreach { faultyDirective =>
+        val inputs = TestInputs(
+          Seq(
+            os.rel / "InvalidUsingDirective.scala" ->
+              s"""$faultyDirective
+                 |
+                 |object InvalidUsingDirective extends App {
+                 |  println("Hello")
+                 |}
+                 |""".stripMargin
+          )
+        )
+        withBsp(inputs, Seq(".")) {
+          (root, _, remoteServer) =>
+            async {
+              val buildTargetsResp = await(remoteServer.workspaceBuildTargets().asScala)
+              val targets          = buildTargetsResp.getTargets.asScala.map(_.getId).toSeq
+              expect(targets.length == 2)
+
+              val bloopProjectNames = targets.map { target =>
+                val targetUri = TestUtil.normalizeUri(target.getUri)
+                checkTargetUri(root, targetUri)
+                new URI(targetUri).getQuery.stripPrefix("id=")
+              }
+
+              val bloopDir = root / Constants.workspaceDirName / ".bloop"
+              expect(os.isDir(bloopDir))
+
+              bloopProjectNames.foreach { bloopProjectName =>
+                val bloopProjectJsonPath = bloopDir / s"$bloopProjectName.json"
+                expect(os.isFile(bloopProjectJsonPath))
+              }
```

Comment:
We should also test, if `bsp` properly generates error diagnostic.

Reply:
good call, testing for diagnostics made me realise those were broken on the initial run (recovering from errors prevented the silenced errors to be transferred as diagnostics).
ought to be fixed now.